apps/dokploy/components/dashboard/application/preview-deployments/preview-deployment-card.tsx

```diff
@@ -0,0 +1,141 @@
+import { StatusTooltip } from "@/components/shared/status-tooltip";
+import { Badge } from "@/components/ui/badge";
+import { Button } from "@/components/ui/button";
+import { Separator } from "@/components/ui/separator";
+import { Clock, GitBranch, GitPullRequest, Pencil } from "lucide-react";
+import Link from "next/link";
+import { ShowModalLogs } from "../../settings/web-server/show-modal-logs";
+import { DialogAction } from "@/components/shared/dialog-action";
+import { ShowPreviewBuilds } from "./show-preview-builds";
+import { RouterOutputs } from "@/utils/api";
+import { AddPreviewDomain } from "./add-preview-domain";
+import { DateTooltip } from "@/components/shared/date-tooltip";
+
+interface PreviewDeploymentCardProps {
+  appName: string;
+  serverId: string;
+  onDeploymentDelete: (deploymentId: string) => void;
+  deploymentId: string;
+  deploymentUrl: string;
+  deployments: RouterOutputs["deployment"]["all"];
+
+  domainId: string;
+  domainHost: string;
+
+  pullRequestTitle: string;
+  pullRequestUrl: string;
+  status: "running" | "error" | "done" | "idle" | undefined | null;
+  branch: string;
+  date: string;
+  isLoading: boolean;
+}
+
+export function PreviewDeploymentCard({
+  appName,
+  serverId,
+
+  onDeploymentDelete,
+  deploymentId,
+  deployments,
+
+  domainId,
+  domainHost,
+
+  pullRequestTitle,
+  pullRequestUrl,
+  isLoading,
+  status,
+  branch,
+  date,
+}: PreviewDeploymentCardProps) {
+  return (

```

Comment:
I feel is better to just pass the deploymentId and the serverId, and then inside the component use this hook


`
const { data: previewDeployment } = api.previewDeployment.one.useQuery({
    previewDeploymentId: previewDeploymentId,
});
  `
  In this way we avoid passing too many props, and thus isolate the component in a more elegant way.

Reply:
i think we must left `onDeploymentDelete` because we need the to refetch deployments when deleting one & `isLoading` on deleting.

Reply:
I think we can also abstract and put the `handleDeletePreviewDeployment` inside `the preview-deployment-card.tsx`

Reply:
I've made the changes, and all that's left is the props, `serverId`, `previewDeploymentId`, and I've added `refetchPreviewDeployments`, to refresh the page when a pr get deleted. thats the only way that i found to make everything work 😣